### PR TITLE
Feature/add crossword status checker

### PR DESCRIPTION
**Pull request by destrim**

### CODE REVIEW

**Comment by destrim** [comment]
Work in progess, check and review functionalities, not code @krzwid @Przemyslaw5

File: lib/view/save_crossword.dart [hunk 1]
```diff
@@ -44,25 +48,105 @@ class _SaveCrossword extends State<SaveCrossword> {
...
+    );
+  }
+
+  TextButton skipCrosswordButton() {
+    return TextButton(
+      style: ButtonStyle(
+        foregroundColor: MaterialStateProperty.all<Color>(Colors.grey),
+      ),
+      onPressed: () async {
+        Navigator.pop(context, true);
```
Comment (Przemyslaw5) [inline lib/view/save_crossword.dart:96]: Save also this image in db.

**Review by Przemyslaw5 — CHANGES_REQUESTED** [review]

File: lib/view/save_crossword.dart (hunk 1 above)
Comment (destrim) [reply lib/view/save_crossword.dart:96]: Done

File: lib/view/my_account.dart [hunk 2]
```diff
@@ -84,4 +92,79 @@ class MyAccountState extends State<MyAccount> {
       (Route<dynamic> route) => false,
     );
   }
+
+  // TODO testButton and its test function only for testing purposes
+  Container testButton() {
```
Comment (Przemyslaw5) [inline lib/view/my_account.dart]: I think these changes are unnecessary. Propably they were only for testing and can be removed now

File: lib/database/crossword_info_repository.dart [hunk 3]
```diff
@@ -32,16 +32,17 @@ class CrosswordInfoRepository {
...
+        where: '${CrosswordInfoFields.id} = ?', whereArgs: [crosswordInfo.id]);
   }
 
-  Future<List<CrosswordInfo>> getAllCrosswordsInfo() async {
+  Future<List<CrosswordInfo>> getAllCrosswordsInfo(String userId) async {
     final db = await CrosswordDatabase.instance.database;
 
-    final result = await db.query(tableName);
+    final result = await db.query(tableName, where:
+    '${CrosswordInfoFields.userId} = ?', whereArgs: [userId]);
```
Comment (Przemyslaw5) [inline lib/database/crossword_info_repository.dart]: Add also where clause to query only crosswordInfo with status `solved_waiting` and `solved_accepted`.

File: lib/util/http_util.dart [hunk 4]
```diff
@@ -3,42 +3,50 @@ import 'dart:io';
...
     var url = Config.makeUriQuery(baseUrl, '/api/solver');
     var request = http.MultipartRequest('POST', url);
 
+    final byteData = await rootBundle.load('assets/images/39.png');
+    var imageFile =
+        File('${(await getTemporaryDirectory()).path}/crossword.jpg');
+    imageFile = await imageFile.writeAsBytes(byteData.buffer
+        .asUint8List(byteData.offsetInBytes, byteData.lengthInBytes));
+
+    // var imageFile = File(imagePath);
```
Comment (Przemyslaw5) [inline lib/util/http_util.dart]: Before merging comment and leave this in code, plaese

File: lib/view/solve_crossword.dart [hunk 5]
```diff
@@ -125,7 +126,8 @@ class TakePictureScreenState extends State<TakePictureScreen> {
     );
   }
 
-  Future<void> choosePhoto(BuildContext context) async {
+  // metody zwracaja tylko cropped image i pozniej solve wywolac osobno
```
Comment (Przemyslaw5) [inline lib/view/solve_crossword.dart]: This comment can be misleading. Fix or delete it.

File: lib/view/solve_crossword.dart [hunk 6]
```diff
@@ -221,9 +271,100 @@ class TakePictureScreenState extends State<TakePictureScreen> {
...
+    print("checkCrosswordStatus");
+
+    var resposne =
+        await HttpUtil.getCrosswordStatus(userId, crosswordId.toString());
+
+    var decodedResponse = jsonDecode(resposne.body);
+    print(decodedResponse);
+
+    var status = decodedResponse['status'];
+    print(status);
```
Comment (Przemyslaw5) [inline lib/view/solve_crossword.dart]: I think you can remove this. In loop we have whole logic.

File: lib/view/solve_crossword.dart [hunk 7]
```diff
@@ -221,9 +271,100 @@ class TakePictureScreenState extends State<TakePictureScreen> {
...
+      context, dynamic decodedResponse, String path, String id, String name) {
+    var status = decodedResponse['status'];
+    var message = decodedResponse['message'];
+
+    return SnackBar(
+      behavior: SnackBarBehavior.floating,
+      content: Text("Status: $status, message: $message"),
+      backgroundColor: (Colors.green),
+      action: SnackBarAction(
+        label: 'Go to crossword',
```
Comment (Przemyslaw5) [inline lib/view/solve_crossword.dart]:
```suggestion
      content: Text("Pomyślnie rozwiązano krzyżówkę"),
      backgroundColor: (Colors.green),
      action: SnackBarAction(
        label: 'Pokaż',
```
And change something with this blue text. Maybe the good idea is to set bold on this text?

File: lib/view/solve_crossword.dart [hunk 8]
```diff
@@ -221,9 +271,100 @@ class TakePictureScreenState extends State<TakePictureScreen> {
     crosswordInfoRepository.insertCrosswordInfo(crosswordInfo);
   }
 
-  void navigateToApp(context) {
-    Navigator.of(context).push(MaterialPageRoute(
-      builder: (context) => const App(),
-    ));
+  SnackBar serverErrorSnackBar(String statusCode) {
```
Comment (Przemyslaw5) [inline lib/view/solve_crossword.dart]: We use this snackbar anywhere?

File: lib/view/solve_crossword.dart [hunk 9]
```diff
@@ -221,9 +271,100 @@ class TakePictureScreenState extends State<TakePictureScreen> {
...
+    );
+  }
+
+  SnackBar serverSolvingErrorSnackBar(dynamic decodedResponse) {
+    var status = decodedResponse['status'];
+    var message = decodedResponse['message'];
+
+    return SnackBar(
+      behavior: SnackBarBehavior.floating,
+      content: Text("Status: $status, message: $message"),
```
Comment (Przemyslaw5) [inline lib/view/solve_crossword.dart]: Please polish language, and write simple mapper english -> polish for message

**Review by Przemyslaw5 — CHANGES_REQUESTED** [review]

File: lib/view/my_account.dart (hunk 2 above)
Comment (destrim) [reply lib/view/my_account.dart]: yup I ll remove it

File: lib/util/http_util.dart (hunk 4 above)
Comment (destrim) [reply lib/util/http_util.dart]: ok, my mistake forgot to comment this

File: lib/view/solve_crossword.dart (hunk 5 above)
Comment (destrim) [reply lib/view/solve_crossword.dart]: gonna fix it

File: lib/view/solve_crossword.dart (hunk 9 above)
Comment (destrim) [reply lib/view/solve_crossword.dart]: ok I will do it

File: lib/view/solve_crossword.dart (hunk 8 above)
Comment (destrim) [reply lib/view/solve_crossword.dart]: We will. When we will do a refactor, all logs with server errors will be replaced with this snackbar showing server error to the user.

**Review by Przemyslaw5 — APPROVED** [review]

**Pull request merged**